examples/example-react-native-wallet/screens/MobileWalletAdapterEntrypointBottomSheet.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {StyleSheet, View, BackHandler, ActivityIndicator} from 'react-native';
+import Modal from 'react-native-modal';
+import {
+  MobileWalletAdapterServiceEventType,
+  MobileWalletAdapterConfig,
+  useMobileWalletAdapterRequest,
+  SessionTerminatedRequest,
+  MobileWalletAdapterServiceRequest,
+  AuthorizeDappRequest,
+  SignPayloadsRequest,
+  MobileWalletAdapterServiceRemoteRequest,
+  SignAndSendTransactionsRequest,
+} from '@solana-mobile/mobile-wallet-adapter-walletlib';
+
+import AuthenticationScreen from '../bottomsheets/AuthenticationScreen';
+import LoadingScreen from '../bottomsheets/LoadingScreen';
+import SignAndSendTransactionsScreen from '../bottomsheets/SignAndSendTransactionsScreen';
+import SignPayloadsScreen from '../bottomsheets/SignPayloadsScreen';
+import WalletProvider from '../components/WalletProvider';
+
+function getRequestScreenComponent(
+  request: MobileWalletAdapterServiceRequest | null | undefined,
+) {
+  switch (request?.type) {
+    case MobileWalletAdapterServiceEventType.SignAndSendTransactions:
+      return (
+        <SignAndSendTransactionsScreen
+          request={request as SignAndSendTransactionsRequest}
+        />
+      );
+    case MobileWalletAdapterServiceEventType.SignTransactions:
+    case MobileWalletAdapterServiceEventType.SignMessages:
+      return <SignPayloadsScreen request={request as SignPayloadsRequest} />;
+    case MobileWalletAdapterServiceEventType.AuthorizeDapp:
+      return <AuthenticationScreen request={request as AuthorizeDappRequest} />;
+    default:
+      return <ActivityIndicator size="large" />;
+  }
+}
+
```

Comment:
This is the MWA Entrypoint component. It is what will be rendered when the user intent is accepted by the wallet app.